### ISSUE TITLE
RFE: poll on youtube alternatives, such as odysee, dlive, rumble etc

### ISSUE THREAD

**Issue by herrdeh** [TeamNewPipe/NewPipe#6645]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

With more and more censorship applied on youtube, we less and less can use it as a valuable source of information. More and more valuable sources evade to other platforms, such as rumble, odysee, dlive - and probably quite some other ones I'm not even aware of. I suggest to hold a poll which alternative video platforms should be supported in future, giving developers an estimation on priorities.

#### Is your feature request related to a problem? Please describe it

- Performance of video platforms on browsers, especially on an older phone, can really be poor. 
- Beloved features, like audio playback with screen turned off, are mostly not available.
- Administration of bookmarks over different tools is no fun.
- Guess, the author of #5889 has the same intention as I have.

#### Additional context

On the long run, my hope is that semi-monopolies like youtube will not survive. And I hope that NewPipe will encourage this process of independence from big tech companies - towards more variety, resilience and competition.
There are some related requests: #4733, #4854, #6031 and #434. These are focussed on adding lbry (odysse), but I think there are more platforms to consider.  

#### How will you/everyone benefit from this feature?

Breaking the youtube semi-monopoly is a value of its own - and NewPipe team can help!

**Comment by evermind-zz** [comment]
You're on the wrong project here! The support for platforms is not in the NewPipe app but inside the NewPipeExtractor lib. For information about supporting other platforms look at the NewPipeExtractor issue tracker for example:  https://github.com/TeamNewPipe/NewPipeExtractor/pull/325#issuecomment-626801193

In short the NewPipe Team is not willing to add alternatives they think are places of misinformation.

Nevertheless I've added support for rumble and reworked support from @yashx [bitchute](https://github.com/yashx/NewPipeExtractor/tree/bitchute) support into a forked [NewPipeExtractor](https://github.com/evermind-zz/NewPipeExtractor). But the moment I have no time to create a fork that leverages on this library.

**Comment by opusforlife2** [comment]
@evermind-zz Although you're right, any poll would be better off hosted on the far more well known NewPipe repo, where more users could respond.

**Comment by herrdeh** [comment]
> In short the NewPipe Team is not willing to add alternatives they think are places of misinformation.

Since beginning of 2020 very much of what we used to regard as the "good guys" and "bad guys" reverted; this makes things really difficult. I'm very much certain - you've got more than a fair chance to overcome those how did not yet understand that y**t*b* changed into a hoard of censorship, government propaganda and and big pharma lobbying.
I very much appreciate your work and I'm looking forward to your fork, please keep us informed!

**Comment by CharString** [comment]
@herrdeh FLOSS is so nice; people can fork their own alternative for consuming alternative facts. Nothing is too niche.

**Comment by evermind-zz** [comment]
> > In short the NewPipe Team is not willing to add alternatives they think are places of misinformation.
> 
> Since beginning of 2020 very much of what we used to regard as the "good guys" and "bad guys" reverted; this makes things really difficult. I'm very much certain - you've got more than a fair chance to overcome those how did not yet understand that y**t_b_ changed into a hoard of censorship, government propaganda and and big pharma lobbying.
> I very much appreciate your work and I'm looking forward to your fork, please keep us informed!

https://github.com/bravenewpipe/NewPipe/releases/tag/v0.21.8-1.0.0

**Comment by degarb** [comment]
Yeah, some of my chemistry, biochemisty, and nuclear professors and doctors that I follow on youtube have been getting strikes and kicked off of youtube because they are finding solutions to problems that is ahead of the regulators in power, including the cdc.

The youtube moderators are way out of their depth, and turning everything into a political discussion, including science and simple library newspaper research into old news articles that is ahead of their own world, political views.  

If the youtube moderators are so antiscience and simple minded, that they want to yank their own platform for an openforum where brainstorming and truth can rise to the top  (for example, in any debate between a physicist and a 911 conspiracy expert, the physicist always wins. And, it is very important that the questions that the 911 conspiracy expert be raised.), then, I will be more confortable moving off of youtube as a user and to a platform that only censors hatred, violence, and keeps a family friendly atmosphere (this is not only because I have kids, but I do not have the time for vulgarity and do not think social media as appropriate for gratuitous sexual manipulation of traffic, where there are alternatives that can focus using that. Because, I view youtube and the likes as an idea exchange (learn to build a house with all aspects and intricacies covered by the right way, the wrong way as pointed out by some of the comments, and the alternative way of doing things).  Note that by doing something the wrong way, that youtuber might open up ideas for a better way of doing something which has never been thought of before.

Yes, it might be time to abandon youtube, if their moderators are knocking us back to the dark ages. 

Interestingly, one wing will always adhere to sex/drug idea freedom, while screaming disinformation when a political idea rubs them the wrong way and might require a deep personal inner change if proved true, while the other  world view would be willing to give up my vulgarity and sexual filters, if it meant allowing people with better ways to build the next car or house, beat disease, etc.  Of course, this is why parental filters were invented with varying strength levels. These filters deprived the viewer of content, not the creators, and not harming the forum in general.  Parental filters can be lowered or removed by the end user, if the end user feels useful content is being block.  There is no reason why a politically set moderator or user could not use those same controls to blind themselves from ideas that rock their world view that are now used to control various degrees of pornography and fictitious violence (like gratuitous violence and sex scenes in most of the Netflix-minus the shows they have for under age 8--  and other's streaming library)...  I write this, in hopes of widening and promoting truth for future generations, and awakening people to the new Spanish inquisition moderators that is killing science, in favor of a very chinese model of censorship to drive a very political agenda and kill off the opposition.  I am shocked that the newpipe team might buy into the empty argument of "misinformation", which is any information that you disagree with.  In truth, everyone, at every point in human history, has been both right and wrong about some things.  There is not one leader, not one vote, not one consensus, not one textbook, not one professor, not one science theory, not one doctor, not one panel of doctors, who have been right about everything.  In fact, History tells us, that they are usually wrong about most things, and their wrong ideas will take generations to over turn with a better theory, which can only happen if the better idea is allowed to be raised.   Censorship to blind one's self, or those around you, is just censorship, it is not "misinformation" suppression any more than it is is Truth suppression.  This is because History tells us that mere humans do not know the Final Truth.  None of us are smart enough--not alone, nor collectively.  This is why Physics is not the same in 1880 as 1980.  Nor is it the same in 2020.  Nor will it be that same in 2080.  Humanity, we know historically, has never had a handle on the truth. All we have is a forum of ideas, some of which hold up better to other to deep scrutiny--that is, if scrutiny and questioning is allowed. As questions arise and are explained, it either weakens or strengthens the consensus. But when you put up a panel of consensus censors (exactly how North Korea became a dictatorship by the Chairman of the Censorship board), the truth never comes out, while the chairman of the censorship board eventually derives absolute power, eventually having unlimited power.  Those on the outside call it a dictatorship (outside north korea), while those on the inside believe they are being sheltered from misinformation and dangerous ideas..... Yeah, by supporting other platforms, you are protecting youtube from being abused by moderators with very little regard for the scientific process and Critical thinking and discussion, as well as undue influence by money coming from political parties and foreign dictatorships (via companies dependent on those markets and cheap manufacturing centers which allow price undercutting).  There is every indication of these influences, which are not friendly in the slightest to Wester ideas of Constitutional Democracy or personal freedom. Rightly, or wrongly, this is were we are today. 

It is a sad day for me, 2 days ago, when a theoretical biochemisty mechanism discussion gets a youtube strike, because they are ahead of a China and big pharmaceutical company led panel throwing out limited recommendations that fall short of solving human health problems.  It shows a total ignorance of Scientific history.

**Comment by degarb** [comment]
I will add that I cannot move off from youtube as a platform without newpipe.

1. I need to be able to listen to scores of hours of lecture audio when I driving or manually working.  Else I cannot keep up and use the half wasted hours.
2. I can learn 50 percent more by watching at 2.7 speed, with silence taken out (some closed captions to aid in missed word understanding).  Since the cell phone allows a form factor more suited to more situations than my laptop, newpipe is the only tool I have.
3. In early 2018, the ads on you tube were tolerable.  But as new android versions arrived the ads started to choke out the content (just like on am radio, which I had to abandon because of the ads back in the 90s).  Newpipe and brave are the only way I can stomach watching youtube anymore on the android platform.

Odysee feels better to me than bitcute.  Bitcute lacked too many things that youtube had.  I hope that I can somehow migrate away from youtube to odysee.  I do not see how I can do it without NewPipe.    If Newpipe team can't see the danger of youtub's double standards and cherry picking censorship, I don't know what to do, because I am not a computer programmer. 
I do see it as essential for our children's future, that newpipe support other platforms than YT.

**Comment by vehosah** [comment]
Like youtube is not full of misinformation 😂

**Comment by degarb** [comment]
The problem is that No mere mortal knows what is truth and what is misinformation.  It is  misinformation to blankly declare one idea as absolute misinformation.

We only have some ideas that are more provable than other idea. However, either the most provable idea will prevail, or the most profitable idea will prevail.  The false disprovable, but very lucrative idea will always prevail, and all competing ideas silenced by declaring them misinformation.   For a time, at least.

**Comment by alxlg** [comment]
@evermind-zz 

> In short the NewPipe Team is not willing to add alternatives they think are places of misinformation.

Thanks for letting us know, if this is true and NewPipe developers would refuse contributions about alternative platforms support then I think it's important to stress this concept more:

Letting people choose what to watch means respect for freedom, humanity, democracy and decentralization of power. Playing the role of good father that choose that kids can see is disrespectful for humans, implying their are not able to discern good and bad information.

Also, historically all improvements humanity had about freedom and other human rights came from a minority opposer by the majority. Be aware that by promoting a mainstream platform and discourage others because of bias directly affect the natural process of improvement of humanity.

**Comment by harmin-parra** [comment]
I would love to have this functionality implemented.

As YouTube decided to do censorship, some videos channels get banned from YouTube, which force video creators to publish their videos in other websites.

**Comment by harmin-parra** [comment]
@evermind-zz 

> In short the NewPipe Team is not willing to add alternatives they think are places of misinformation.

To my knowledge, Youtube has no problem allowing old videos about the false Iraq weapons of mass destruction claims that lead to that war.
Isn't that misinformation from the part of YouTube ? LOL

Well, I didn't know that the NewPipe Team was also going to decide for us what is misinformation and what is not. LOL

**Comment by degarb** [comment]
If YouTube was honest they would just admit that 60 percent of their advertising came from big drug markers, insurance companies, and the government funded media blitz of the NIH, niaid, and rest of their money from China and the NIH.  

NewPipe is not supporting science by supporting youtu.be, they are supporting the advertisers and the misinformation narrative to ensure the 200+ billion dollar vacancies got approval and paid for. I am very pro vaccine, but the unnecessary body count due to information suppression was on the level of the old Stalin Soviet union, as our democracy tried to become more like China.

**Comment by litetex** [comment]
I don't think this issue is currently very helpful and won't be so in the future.

Why?
* There is a lot of off-topic discussion about YT. If you annoyed about what YT does, discuss it somewhere else - we just implement/provide it as a source. 
  * I also noticed that those off-topic conversations do not cite any sources
* You failed to realize that NewPipe is already (partial) controlled by the userbase. GitHub e.g. allows to give issues upvotes or let the user create a PR. The userbase is also mainly using Youtube. Where do I know that from? Well most issues that we get are about YT:

| Service/Keyword inside issue | Open and closed issues updated since 2021 | Percentage of issues |
| --- | --- | --- |
| YT | [284+652](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aopen+youtube+updated%3A%3E%3D2021-01-01+) | 84% |
| Soundcloud | [33+44](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aopen+soundcloud+updated%3A%3E%3D2021-01-01+) | 7% |
| media.ccc.de | [5+9](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aopen+media+ccc+updated%3A%3E%3D2021-01-01+) | 1% |
| Peertube | [21+35](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aopen+peertube+updated%3A%3E%3D2021-01-01+) | 5% |
| Bandcamp| [14+14](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aopen+bandcamp+updated%3A%3E%3D2021-01-01+) | 3% |

* There are a ton of [service requests inside NewPipe Extractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues?q=is%3Aopen+is%3Aissue+label%3A%22new+service%22) some are open since a long time, however nobody implements them - which is likely caused by the fact that not much people care about them
* The "beta" services are already poorly maintained and I think adding more services would make the situation worse

TL;DR 
I see no reasons for a vote and what it should be good for.
I'm to 100% sure that if we now start a vote here it would be distorted by a crying minority and most people - who are fine with NewPipe - won't vote.
Similar patterns can be observed when we e.g. switch to a newer Android version and drop support for a outdated one: Most people don't care because they are up-to-date but a small amount of users who would be affected starts a comment war in the issue.

**Comment by Lee-Carre** [comment]
> There is a lot of off-topic discussion about YT. If you annoyed about what YT does, discuss it somewhere else - we just implement/provide it as a source.

Yet, that sentiment somehow (magically) doesn't apply when it comes to Bitchute. Funny, that.

People are only going to take policy-sounding or policy-like statements when they're (internally-)consistent, and (intellectually-)honest.

Re ‘if you want it then implement it yourself’.

People [tried, and those PRs were rejected](/TeamNewPipe/NewPipeExtractor/pull/325#issuecomment-626801193).

Besides, there are far more users than developers.

The argument about YT being popular is potentially self-reinforcing (if not self-fulfilling). Chicken & egg problem.

People know that they can likely rely on NewPipe for YT access, but evidently have to use other software for accessing non-legacy video-hosting sites (other than PeerTube).

Collecting opinions of interested users is useful to know which features are desired, to keep development relevant.

If NewPipe isn't about being useful software to real-world users (as inconvenient as humans may be, sometimes), then what *is* its purpose?

**Comment by litetex** [comment]
> > There is a lot of off-topic discussion about YT. If you annoyed about what YT does, discuss it somewhere else - we just implement/provide it as a source. 
>
> Yet, that sentiment somehow (magically) doesn't apply when it comes to Bitchute. Funny, that.

I meant the discussions here and not the content of the site.
Btw, I just opened the bitbuchte homepage and >90% of all videos I saw are about conspiracy theories (mostly COVID). So I totally understand why the implementation was declined.


> Besides, there are far more users than developers.

*Unexpected*

> People know that they can likely rely on NewPipe for YT access, 

As I mentioned above: Most users are using YT therefore YT is the most maintained service.


> but evidently have to use other software for accessing non-legacy video-hosting sites (other than PeerTube).

I'm not undestanding what this is about but yes if the service is not implemented in NewPipe you can't use it.


> Collecting opinions of interested users is useful to know which features are desired, to keep development relevant.

As I mentioned above this is e.g. done by GitHub upvotes.


> If NewPipe isn't about being useful software to real-world users (as inconvenient as humans may be, sometimes), then what is its purpose?


I can't see how this is related to my above comment.

**Comment by opusforlife2** [comment]
> but developers decided they want to boycott those platforms because of personal beliefs.

Really? From the Readme of https://github.com/bravenewpipe/NewPipe:
> platforms that promote pornography or other degrading things will NOT be included here

Hypocrisy much?

> any FOSS project is controlled by the userbase, come on

Never forget Gab: [How the biggest decentralized social network is dealing with its Nazi problem](https://www.theverge.com/2019/7/12/20691957/mastodon-decentralized-social-network-gab-migration-fediverse-app-blocking)

----

We apologise that this issue stayed open for so long and encouraged all the off-topic comments. There was never any point in commenting here. We debate the merits of adding or refusing to add a service on a per-service basis, not as a group of services representing a theme or idea.

So if you would like to point out the merits/demerits of a particular service, please go to the issue for _that_ service in the Extractor repo here: https://github.com/TeamNewPipe/NewPipeExtractor/issues. If it doesn't already exist, then create a new one.
